tools/internal/mdocspec/md_test.go

```diff
@@ -0,0 +1,399 @@
+paths:
+  # interface configuration
+  /interfaces/interface/config/description:
+  /interfaces/interface/config/enabled:
+  # name of chassis component
+  /components/component/state/name:
+    platform_type: "CHASSIS"
+
+rpcs:
+  gnmi:
+    gNMI.Set:
+      union_replace: true
+    gNMI.Subscribe:
+      on_change: true
+` + "```" + `
+
+## Required DUT platform
+
+` + "```" + `yaml
+paths:
+  # interface configuration
+  /a/b/c:
+  /d/e/f:
+
+rpcs:
+  fooi:
+    fooi.Set:
+      union_replace: true
+    fooi.Subscribe:
+      on_change: true
+` + "```" + `
+
+* Specify the minimum DUT-type:
+  * MFF - A modular form factor device containing LINECARDs, FABRIC and redundant CONTROLLER_CARD components
+  * FFF - fixed form factor
+  * vRX - virtual router device
+`),
+		want: `paths:
+  # interface configuration
+  /interfaces/interface/config/description:
+  /interfaces/interface/config/enabled:
+  # name of chassis component
+  /components/component/state/name:
+    platform_type: "CHASSIS"
+
+rpcs:
+  gnmi:
+    gNMI.Set:
+      union_replace: true
+    gNMI.Subscribe:
+      on_change: true
+`,
+	}, {
+		desc: "two-yaml-blocks-same-heading",
+		inSource: []byte(`---
+name: New featureprofiles test requirement
+about: Use this template to document the requirements for a new test to be implemented.
+title: ''
+labels: enhancement
+assignees: ''
+---
+
+## Procedure
+
+* Test environment setup
+  * Description of procedure to configure ATE and DUT with pre-requisites making it possible to cover the intended paths and RPC's.
+
+* TestID-x.y.z - Name of subtest
+  * Step 1
+  * Step 2
+  * Validation and pass/fail criteria
+
+* TestID-x.y.z - Name of subtest
+  * Step 1
+  * Step 2
+  * Validation and pass/fail criteria
+
+## OpenConfig Path and RPC Coverage
+
+This example yaml defines the OC paths intended to be covered by this test.  OC paths used for test environment setup are not required to be listed here.
+
+` + "```" + `yaml
+paths:
+  # interface configuration
+  /a/b/c:
+  /d/e/f:
+
+rpcs:
+  fooi:
+    fooi.Set:
+      union_replace: true
+    fooi.Subscribe:
+      on_change: true
+` + "```" + `
+
+` + "```" + `yaml
+paths:
+  # interface configuration
+  /interfaces/interface/config/description:
+  /interfaces/interface/config/enabled:
+  # name of chassis component
+  /components/component/state/name:
+    platform_type: "CHASSIS"
+
+rpcs:
+  gnmi:
+    gNMI.Set:
+      union_replace: true
+    gNMI.Subscribe:
+      on_change: true
+` + "```" + `
+
+## Required DUT platform
+
+* Specify the minimum DUT-type:
+  * MFF - A modular form factor device containing LINECARDs, FABRIC and redundant CONTROLLER_CARD components
+  * FFF - fixed form factor
+  * vRX - virtual router device
+`),
+		want: `paths:
+  # interface configuration
+  /a/b/c:
+  /d/e/f:
+
+rpcs:
+  fooi:
+    fooi.Set:
+      union_replace: true
+    fooi.Subscribe:
+      on_change: true
+`,
+	}, {
+		desc: "two-blocks-same-heading-first-language-not-specified",
```

Comment:
I think I would like to see the language required, just for pretty rendering for humans to read.  WDYT?

Reply:
This is the intended behaviour, improved error desc string to make clearer.